CONTRIBUTING.md

```diff
@@ -1,67 +1,117 @@
+    - [Customizing `config/kibana.dev.yml`](#customizing-configkibanadevyml)
+    - [Setting Up SSL](#setting-up-ssl)
+  - [Linting](#linting)
+  - [Testing and Building](#testing-and-building)
+  - [Debugging Unit Tests](#debugging-unit-tests)
+  - [Unit Testing Plugins](#unit-testing-plugins)
+  - [Running Browser Automation Tests](#running-browser-automation-tests)
+    - [Browser Automation Notes](#browser-automation-notes)
+  - [Building OS packages](#building-os-packages)
+- [Signing the contributor license agreement](#signing-the-contributor-license-agreement)
+- [Submitting a Pull Request](#submitting-a-pull-request)
+- [Code Reviewing](#code-reviewing)
+  - [Getting to the Code Review Stage](#getting-to-the-code-review-stage)
+  - [Reviewing Pull Requests](#reviewing-pull-requests)
+
+Don't fret, it's not as daunting as the table of contents makes it out to be!
+
+## Effective issue reporting in Kibana
+
+At any given time the Kibana team at Elastic is working on dozens of features and enhancements, both for Kibana itself and for a few other projects at Elastic. When you file an issue, we'll take the time to digest it, consider solutions, and weigh its applicability to both the Kibana user base at large and the long-term vision for the project. Once we've completed that process we will assign the issue a priority.
+
+- **P1**: A high-priority issue that affects virtually all Kibana users. Bugs that would cause incorrect results, security issues and features that would vastly improve the user experience for everyone. Work arounds for P1s generally don't exist without a code change.
 - **P2**: A broadly applicable, high visibility, issue that enhances the usability of Kibana for a majority users.
 - **P3**: Nice-to-have bug fixes or functionality. Work arounds for P3 items generally exist.
 - **P4**: Niche and special interest issues that may not fit our core goals. We would take a high quality pull for this if implemented in such a way that it does not meaningfully impact other functionality or existing code. Issues may also be labeled P4 if they would be better implemented in Elasticsearch.
 - **P5**: Highly niche or in opposition to our core goals. Should usually be closed. This doesn't mean we wouldn't take a pull for it, but if someone really wanted this they would be better off working on a plugin. The Kibana team will usually not work on P5 issues but may be willing to assist plugin developers on IRC.
 
-#### How to express the importance of an issue
-Let's just get this out there: **Feel free to +1 an issue**. That said, a +1 isn't a vote. While we keep up on highly +1'd issues, a solid write up of your use case is more likely to make your case than a comment that says *+10000*.
+### Voicing the importance of an issue
+
+While we keep up on highly :+1: and heavily commented issues, a solid write up of your use case is more likely to grab our attention than a ":+1:" _(or even a ":100:"!)_ comment.
```

Comment:
How do you feel about explicitly saying "use issue reactions for +1s, don't write comments with just a +1"? I'd like this section to focus even more on encouraging well written, thoughtful comments. I'd suggest something like this:

"If an issue is important to you, add a comment with a solid write up of your use case and an explanation of why it's so important. Don't just comment with a 👍 . We seriously appreciate thoughtful comments. By sharing your thoughts we might even be able to come up with creative alternative solutions for your specific problem. If everything you'd like to say has already been said but you'd like to add a token of support, feel free to use the 👍  [reaction](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) on the issue itself and the comment which best summarizes your thoughts."